sorting/merge_sort/multithreaded_merge_sort_golang/main.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"fmt"
+)
+
+func parallelMergeSort(arr []int, c chan []int) {
+	if len(arr) <= 1 {
+		c <- arr
+		close(c)
+		return
+	}
+
+	mid := len(arr) / 2
+	leftChan := make(chan []int)
+	rightChan := make(chan []int)
+
+	go parallelMergeSort(arr[:mid], leftChan)
+	go parallelMergeSort(arr[mid:], rightChan)
+
+	left := <-leftChan
+	right := <-rightChan
+
+	c <- merge(left, right)
+	close(c)
+}
+
+func merge(left, right []int) []int {
+	result := make([]int, len(left)+len(right))
```

Comment:
can use a better self-explanatory var name

Reply:
Ok, I will change it